src/reducers/index.ts

```diff
@@ -105,19 +105,31 @@ function parseVega(
   };
 }
 
-function parseVegaLite(
-  state: State,
-  action: SetVegaLiteExample | UpdateVegaLiteSpec | SetGistVegaLiteSpec,
-  extend: Partial<State> = {}
-) {
+function parseVegaLite(state: State, action: any, extend: Partial<State> = {}) {
   const currLogger = new LocalLogger();
 
   try {
+    // if we are setting the spec we don't pass the config
+    // add the previous config from the state
+    if (!action.config) {
```

Comment:
This is causing side effects. Instead, rewrite the `options` code below to use `action.config || state.config`.